docs/advanced-querying-techniques/ctes.md

```diff
@@ -0,0 +1,398 @@
+    FROM bank
+    GROUP BY job
+)
+SELECT * 
+FROM Job_Avg_Balance, Job_Avg_Campaign 
+WHERE Job_Avg_Balance.job = Job_Avg_Campaign.job;
+```
+
+In this example, the first CTE is the same as in the previous example. The second CTE `Job_Avg_Campaign` returns two columns: `job_1` and `AverageCampaign`. The final SELECT statement retrieves data from both CTEs. Notice that the entries.
+
+You will notice `job` and `job_1` in the final result. This is because we are doing a Cartesian product (cross join) between two CTEs and both have a column named `job`. We can avoid this by explicitly specifying the columns you want to select in our final `SELECT` statement instead of using `SELECT *`.
+
+```{code-cell} ipython3
+%%sql
+WITH Job_Avg_Balance AS (
+    SELECT job, AVG(balance) AS AverageBalance
+    FROM bank
+    GROUP BY job
+),
+Job_Avg_Campaign AS (
+    SELECT job, AVG(campaign) AS AverageCampaign
+    FROM bank
+    GROUP BY job
+)
+SELECT Job_Avg_Balance.job, AverageBalance, AverageCampaign
+FROM Job_Avg_Balance
+JOIN Job_Avg_Campaign 
+ON Job_Avg_Balance.job = Job_Avg_Campaign.job;
+```
+
+## Recursive CTEs
+
+A recursive CTE is one in which an initial CTE is repeatedly executed to return subsets of data until the complete result set is obtained.
+
+The given dataset doesn't lend itself to a recursive CTE, as these are generally used for hierarchical or recursive data problems, which the bank dataset does not present.
+
+Here is an example of a recursive CTE
+
+```{code-cell} ipython3
+%%sql
+WITH RECURSIVE numbers AS (
+    SELECT 1 AS value
+    UNION ALL
+    SELECT value + 1 FROM numbers WHERE value < 10
+)
+SELECT * FROM numbers;
+```
+
+This will output a list of numbers from 1 to 10. The CTE works as follows:
+
+* The `WITH RECURSIVE` clause marks the start of the recursive CTE.
+* The `SELECT 1 AS value` is the "anchor member" of the CTE and provides the base result set for the recursion to start.
+* The `UNION ALL` clause is used to combine the results of the anchor member with the results of the "recursive member", which is `SELECT value + 1 FROM numbers WHERE value < 10`.
+* The recursion continues until `value < 10` returns false, at which point the CTE stops executing.
+
+## Using CTEs to modify information in a table
+
+```{important}
+Without a unique identifier for each row in your table, performing UPDATE or DELETE operations using a CTE would be risky because they could affect more rows than you intend. 
+```
+
+It's generally not recommended to use UPDATE or DELETE without a unique identifier or precise condition to pinpoint exactly which rows you want to affect. We're going to load data from the [Joining Data in SQL tutorial](https://ploomber-sql.readthedocs.io/en/latest/intro-to-sql/joining-data-in-sql.html#load-the-data) as this data has tables with unique ID's.
+
+```{code-cell} ipython3
+_ = banking.MarketData("https://tinyurl.com/jb-bank-m", "expanded_data")
+_.extract_asc_to_csv()
```

Comment:
Can hide the ouptut of this cell `:tags: [hide-output]`

Reply:
Added